src/dropdown/dropdown.js

```diff
@@ -170,9 +162,53 @@ angular.module('ui.bootstrap.dropdown', ['ui.bootstrap.position'])
         return;
       }
       dropdownCtrl.dropdownMenu = element;
+
+      var dropdown, appendToBody,
+          dropdownMenu = element;
+
+      scope.$watch(dropdownCtrl.isOpen, function( isOpen ) {
```

Comment:
@wesleycho I believe Link function is more appropriate, because most of the logic here is DOM-related (and Link function and Controller share the same scope in this case).